guide.md

```diff
@@ -0,0 +1,350 @@
+`--config` flag is provided, like so:
+
+```
+build:nix --host_platform=@io_tweag_rules_nixpkgs//nixpkgs/platforms:host
+build:nix --crosstool_top=@nixpkgs_config_cc//:toolchain
+```
+
+None of these options will apply to `bazel build` or `bazel run` unless an
+extra `--config nix` parameter is passed:
+
+```
+> bazel run //src:hello-world --config=nix
+...
+```
+
+# Working from a Nix-provided developer environment
+
+The `nix-shell` command we used to get a Nix-provided Bazel was the simplest
+example of a Nix developer environment, but they can provide developers on a
+complex project with a lot more convenience. We can create development
+environments that all contributors on a project can use with zero setup. These
+shells provide consistent versions of libraries for whichever languages are
+used in the project as well as for system dependencies.
+
+We haven't _disabled_ Bazel's deep-rooted tendency to depend on the global
+environment. Indeed, you can rename `.bazelrc` and rerun `bazel run`, and
+everything will work just fine: Bazel will simply fall back to
+`@local_config_cc`, because a C++ compiler toolchain is still globally
+available.
+
+We can provide a Nix developer environment that hermetically provides the specific
+dependencies that developers need and nothing else:
+
+```nix
+{
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-22.05";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { nixpkgs, flake-utils, ... }:
+    # For every platform that Nix supports, we ...
+    flake-utils.lib.eachDefaultSystem (system:
+      # ... get the package set for this particular platform ...
+      let pkgs = import nixpkgs { inherit system; };
+      in {
+        # ... and define a development shell for it ...
+        devShells.default =
+          # ... with no globally-available CC toolchain ...
+          pkgs.mkShellNoCC {
+            name = "rules_nixpkgs_shell";
+            # ... which makes available the following dependencies, 
+            # all sourced from the `pkgs` package set:
+            packages = with pkgs; [ bazel_5 bazel-buildtools cacert nix git ];
+          };
+      });
+
+}
+```
+
+We've also removed the "default" system-wide CC toolchain: the commonly-used
+`mkShell` function creates a development shell with a given set of available
+dependencies _and_ a default CC toolchain, so we've replaced it with
+`mkShellNoCC`, which, as the name suggests, does the same job but without
+implicitly making a CC toolchain available. This means it's no longer possible
+to unintentionally use `@local_config_cc`, as that toolchain won't be able to
+find the global binaries it wants and will fail to work.
+
+Now developers on different platforms can enter a development environment by
+running `nix develop`, and this environment will be identical across platforms.
+This can be made even more seamless by using [direnv](direnv), which has support for Nix flakes:
```

Comment:
This is the first time you mention flakes explicitly, and it is confusing for someone who is not familiar with Nix, i.e. your target audience IIUC.

Reply:
I reworded this to skip "flakes" and just mention the flake.nix file.

Reply:
I also added a link to our "intro to flakes" blog posts.